src/ray/raylet/worker_pool.cc

```diff
@@ -1306,11 +1311,23 @@ void WorkerPool::PrestartWorkers(const TaskSpecification &task_spec,
     int64_t num_needed = desired_usable_workers - num_usable_workers;
     RAY_LOG(DEBUG) << "Prestarting " << num_needed << " workers given task backlog size "
                    << backlog_size << " and available CPUs " << num_available_cpus;
-    for (int i = 0; i < num_needed; i++) {
-      PopWorkerStatus status;
-      StartWorkerProcess(
-          task_spec.GetLanguage(), rpc::WorkerType::WORKER, task_spec.JobId(), &status);
-    }
+    PrestartDefaultCpuWorkers(task_spec.GetLanguage(), num_needed);
+  }
+}
+
+void WorkerPool::PrestartDefaultCpuWorkers(ray::Language language, int64_t num_needed) {
+  static const WorkerCacheKey kDefaultCpuWorkerCacheKey{/*serialized_runtime_env*/ "",
```

Comment:
Does this mean when we schedule an actor, all these workers will be just killed and GC;ed? 

This can potentially have regression in actor scheduling speed when we use `ray.init` because IIUC after this change, actor scheduling will kill all existing workers and start new ones?

Reply:
there is no behavior change in this PR.

Reply:
Seems like a bug that we prestart worker that won't be fit with the incoming task, file bug https://github.com/ray-project/ray/issues/33621